src/inttest/java/com/faforever/api/avatar/AvatarControllerTest.java

```diff
@@ -0,0 +1,146 @@
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.Sql.ExecutionPhase;
+import org.springframework.test.web.servlet.request.MockMultipartHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Sql(executionPhase = ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:sql/prepDefaultUser.sql")
+@Sql(executionPhase = ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:sql/prepAvatarData.sql")
+@Sql(executionPhase = ExecutionPhase.AFTER_TEST_METHOD, scripts = "classpath:sql/cleanAvatarData.sql")
+public class AvatarControllerTest extends AbstractIntegrationTest {
+
+  @Autowired
+  AvatarRepository avatarRepository;
+
+  @Test
+  @WithUserDetails(AUTH_MODERATOR)
+  public void moderatorCanUpload() throws Exception {
+    mockMvc.perform(
+      createAvatarUploadRequest()
+        .with(getOAuthToken(OAuthScope._UPLOAD_AVATAR))
+    ).andExpect(status().isCreated())
+      .andExpect(content().string(""));
+    final Avatar avatar = avatarRepository.findOneByUrl("http://localhost/faf/avatars/avatar3.png").get();
+    assertThat(avatar.getUrl(), is("http://localhost/faf/avatars/avatar3.png"));
+    assertThat(avatar.getTooltip(), is("Best avatar"));
+  }
+
+  @Test
+  @WithUserDetails(AUTH_MODERATOR)
+  public void moderatorCanReupload() throws Exception {
+    Files.copy(FileHandlingHelper.loadResourceAsStream("/avatars/donator.png"), Paths.get("build/cache/avatars/avatar1.png"));
+    mockMvc.perform(
+      createAvatarReuploadRequest(1)
+        .with(getOAuthToken(OAuthScope._UPLOAD_AVATAR))
+    ).andExpect(status().isOk())
+      .andExpect(content().string(""));
+    final Avatar avatar = avatarRepository.findOneByUrl("http://localhost/faf/avatars/avatar1.png").get();
+    assertThat(avatar.getUrl(), is("http://localhost/faf/avatars/avatar1.png"));
+    assertThat(avatar.getTooltip(), is("Best avatar"));
+  }
+
+  @Test
+  @WithUserDetails(AUTH_MODERATOR)
+  public void moderatorCanDeleteAvatar() throws Exception {
+    Files.copy(FileHandlingHelper.loadResourceAsStream("/avatars/donator.png"), Paths.get("build/cache/avatars/avatar1.png"));
+    mockMvc.perform(
+      delete("/avatars/1")
+        .with(getOAuthToken(OAuthScope._UPLOAD_AVATAR))
+    ).andExpect(status().isNoContent());
+    assertThat(avatarRepository.findById(1), is(Optional.empty()));
+  }
```

Comment:
We should assert, that the file has been remove from file system.

Reply:
this is tested in unit test of AvatarService